tests/serverpod_test_server/test_integration/database_operations/persist/non_persist_field_test.dart

```diff
@@ -0,0 +1,145 @@
+import 'package:serverpod/serverpod.dart';
+import 'package:serverpod_test_server/src/generated/protocol.dart';
+import 'package:serverpod_test_server/test_util/test_serverpod.dart';
+import 'package:test/test.dart';
+
+void main() async {
+  var session = await IntegrationTestServer().session();
+
+  group('Given a class with fields that include non-persisted fields,', () {
+    tearDownAll(() async => ObjectFieldScopes.db.deleteWhere(
+          session,
+          where: (_) => Constant.bool(true),
+        ));
+
+    test(
+      'when inserting a single record into the database, non-persisted fields should retain their values after insertion, even though they are not stored in the database',
```

Comment:
NIT: add `then` instead of comma, applies to all tests
```suggestion
      'when inserting a single record into the database then non-persisted fields should retain their values after insertion, even though they are not stored in the database',
```